File: app/src/main/java/com/fengniao/news/ui/activity/MainActivity.java
package com.fengniao.news.ui.activity;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;

import com.fengniao.news.R;
import com.fengniao.news.ui.base.BaseActivity;
import com.fengniao.news.ui.fragment.TouTiaoNewsFragment;
import com.fengniao.news.ui.fragment.ZhiHuArticleFragment;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

public class MainActivity extends BaseActivity {
    @BindView(R.id.tab_layout)
    TabLayout tabLayout;
    @BindView(R.id.view_pager)
    ViewPager viewPager;

    private String[] titles = {"知乎日报", "今日头条"};
    private List<Fragment> mList;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    public void initView() {
        mList = new ArrayList<>();
        mList.add(new ZhiHuArticleFragment());
        mList.add(new TouTiaoNewsFragment());
        MyViewPagerAdapter mAdapter = new MyViewPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(mAdapter);
        viewPager.setOffscreenPageLimit(mList.size());
        tabLayout.setupWithViewPager(viewPager);
    }


    private class MyViewPagerAdapter extends FragmentStatePagerAdapter {
        MyViewPagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            return mList.get(position);
        }

        @Override
        public int getCount() {
            return mList.size();
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return titles[position];
        }
    }
}

File: app/src/main/java/com/fengniao/news/ui/activity/MainActivity.kt
package com.fengniao.news.ui.activity

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.TabLayout
import com.fengniao.news.R
import com.fengniao.news.base.BaseActivity
import com.fengniao.news.module.NewsFragment
import com.fengniao.news.module.read.YueFMActivity
import com.fengniao.news.module.test.MyTestActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : BaseActivity() {
    override fun getLayoutId(): Int = R.layout.activity_main

    lateinit var tabLayout: TabLayout
    var newsFragment: NewsFragment? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initView()
//        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.READ_PHONE_STATE)
//                != PackageManager.PERMISSION_GRANTED) {
//            ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.READ_PHONE_STATE), 1)
//        } else
//            Log.i("test", Utils.getDeviceId(this))
    }

//    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
//        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
//        if (requestCode == 1 && permissions.isNotEmpty()) {
//            Log.i("test", Utils.getDeviceId(this))
//        }
//    }


    private fun initView() {
        supportActionBar?.setDisplayHomeAsUpEnabled(false)
        tabLayout = findViewById(R.id.tab_layout) as TabLayout
        initHomePageFragment()
//        drawer_layout.addDrawerListener(object : DrawerLayout.DrawerListener {
//            //drawer滑动时多次调用
//            override fun onDrawerSlide(drawerView: View?, slideOffset: Float) {
//            }
//            //drawer彻底关闭后调用一次
//            override fun onDrawerClosed(drawerView: View?) {
//            }
//            //drawer彻底打开后调用一次
//            override fun onDrawerOpened(drawerView: View?) {
//            }
//            //drawer状态发生变化时，和viewpager类似，state为1时表示drawer正在拖动，
//            //state为2是表示drawer滑动到最终位置
//            //state为0时表示drawer完成动作处于空闲
//            //操作顺序为1——》2——》0
//            override fun onDrawerStateChanged(newState: Int) {
//            }
//
//        })
        navigation_view.setNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.item_news -> initHomePageFragment()
                R.id.item_read -> {
                    startActivity(Intent(this, YueFMActivity::class.java))
                }
                R.id.item_test -> {
                    startActivity(Intent(this, MyTestActivity::class.java))
                }
            }
            drawer_layout.closeDrawers()
            false
        }
        app_bar_layout.addOnOffsetChangedListener { appBarLayout, verticalOffset -> }
    }

    private fun initHomePageFragment() {
        newsFragment = supportFragmentManager.findFragmentByTag("home") as NewsFragment?
        if (newsFragment == null)
            newsFragment = NewsFragment()
        if (newsFragment!!.isVisible) return
        if (newsFragment!!.isAdded) {
            supportFragmentManager.beginTransaction().show(newsFragment).commit()
        } else
            supportFragmentManager.beginTransaction()
                    .add(R.id.content, newsFragment, "home")
                    .show(newsFragment)
                    .commit()
    }

}
